imap_processing/tests/mag/test_mag_l1c.py

```diff
@@ -1,40 +1,292 @@
+from pathlib import Path
+
 import numpy as np
 import pytest
 import xarray as xr
+import yaml
 
-from imap_processing.mag.l1c.mag_l1c import mag_l1c
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.mag.l1c.interpolation_methods import InterpolationFunction
+from imap_processing.mag.l1c.mag_l1c import (
+    fill_normal_data,
+    find_all_gaps,
+    find_gaps,
+    generate_timeline,
+    interpolate_gaps,
+    mag_l1c,
+    process_mag_l1c,
+)
+from imap_processing.tests.mag.conftest import mag_l1a_dataset_generator
 
 
 @pytest.fixture(scope="module")
 def mag_l1b_dataset():
-    epoch = xr.DataArray(np.arange(20), name="epoch", dims=["epoch"])
-    direction = xr.DataArray(np.arange(4), name="direction", dims=["direction"])
-    vectors = xr.DataArray(
-        np.zeros((20, 4)),
-        dims=["epoch", "direction"],
-        coords={"epoch": epoch, "direction": direction},
-    )
+    output_dataset = mag_l1a_dataset_generator(10)
 
+    output_dataset["epoch"] = xr.DataArray(
+        np.arange(0.1, 5.1, step=0.5) * 1e9, name="epoch", dims=["epoch"]
+    )
+    vectors = np.array([[i, i, i, 2] for i in range(1, 11)])
     vectors[0, :] = np.array([1, 1, 1, 0])
+    output_dataset["vectors"].data = vectors
+
+    return output_dataset
+
+
+@pytest.fixture()
+def norm_dataset():
+    dataset = mag_l1a_dataset_generator(10)
+    epoch_vals = np.array([0, 0.5, 1, 1.5, 2, 4, 4.25, 5.5, 5.75, 6]) * 1e9
+    vectors_per_second_attr = "0:2,4000000000:4"
```

Comment:
I would be tempted to have a method like `generate_science_epochs(how_many_seconds, vectors_per_second)` and a constant say `FOUR_VECTORS_PER_SECOND = 4` to make very explicit what you are using to test and reduce the duplication accross tests (from experience you end up using it a lot) and then you can generate any length of data at any cadence.